iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/utils/HelmFilesystemUtils.java

```diff
@@ -111,4 +108,31 @@ private static String resolveToInputFile(Path helmDirectoryPath, InputFile addit
   public static String normalizeToUnixPathSeparator(String filename) {
     return filename.replace('\\', '/');
   }
+
+  public static String normalizeToRuntimePathSeparator(String filename) {
+    return filename.replace('\\', File.separatorChar)
+      .replace('/', File.separatorChar);
+  }
+
+  public static boolean includeFile(String name) {
+    if (INCLUDED_EXTENSIONS.stream().anyMatch(name::endsWith)) {
+      var parent = new File(".");
+      var child = new File(parent, name);
+
+      boolean isPathSecure;
+      try {
+        isPathSecure = child.getCanonicalPath().startsWith(parent.getCanonicalPath() + File.separator) &&
```

Comment:
What if you have a directory named `aa..` or `..bb`? OK, it's a rare case similar to `~/file.txt` or `.` (current directory).
I just found that similar code mitigates the attack, because `File.getCanonicalPath()` returns absolute path without `..` and others and it's system-depended.

Reply:
If you still think that simple looking for `../`, `/..` and others is better I'm ok to change it.

Reply:
I'm not sure it's better, I wasn't sure about the added benefit. But if it's an added layer of security and removes edge cases were unsure of, I think it's ok to use it.